src/components/app/room/index.tsx

```diff
@@ -0,0 +1,42 @@
+import { Component, ComponentDidLoad, Event, EventEmitter, h, Prop } from "@stencil/core"
+if (!(globalThis as any).URLPattern)
+	(async () => await import("urlpattern-polyfill"))()
+
+@Component({
+	tag: "smoothly-app-room",
+	styleUrl: "style.css",
+	scoped: true,
+})
+export class SmoothlyAppRoom implements ComponentDidLoad {
+	@Prop() label?: string
+	@Prop() icon?: string
+	@Prop() path: string | URLPattern = ""
+	@Prop() to?: string
+	@Prop({ reflect: true }) selected?: boolean
+	@Event() smoothlyRoomSelected: EventEmitter<HTMLElement>
+	contentElement: HTMLElement
+
+	componentDidLoad(): void | Promise<void> {
+		if ((typeof this.path == "string" ? new URLPattern({ pathname: this.path }) : this.path).test(window.location))
+			this.smoothlyRoomSelected.emit(this.contentElement)
+	}
+	render() {
+		return [
+			<li>
+				<a
+					href={typeof this.path == "string" ? this.path : this.path.pathname}
+					onClick={(event: PointerEvent) => {
+						if (!event.metaKey && !event.ctrlKey && event.which != 2 && event.button != 1) {
```

Comment:
why is this needed?

Reply:
we need a check statement to see if the specific keys are pressed with the mouse click as well, and we want to prevent that from firing our event